packages/http-client/src/http-client.ts

```diff
@@ -176,18 +203,20 @@ async function getMergedHeaders(
   });
 }
 
-function getMergedParameters(
+function getMergedAndEncodedParameter(
   destination: Destination,
   paramsOriginOptions?: OriginOptions
-): Record<string, string> | undefined {
+): Record<string, any> | undefined {
   const queryParametersDestinationProperty = getAdditionalQueryParameters(
     (destination.originalProperties as DestinationConfiguration) || {}
   ).queryParameters;
-  return mergeOptionsWithPriority({
+  const parametersFromAllOrigins = {
     ...paramsOriginOptions,
     destinationProperty: queryParametersDestinationProperty,
     destination: destination.queryParameters
-  });
+  };
+  const encodedParameters = getEncodedParameters(parametersFromAllOrigins);
+  return mergeOptionsWithPriority(encodedParameters);
```

Comment:
Should you encode first and merge after? I would assume the other way around would be better (avoid unnecessary encoding, although unlikely it could prevent issues with merging caused by encoding)